src/fmi4c.c

```diff
@@ -422,7 +422,6 @@ bool parseModelDescriptionFmi2(fmiHandle *fmu)
                 free((char*)causality);
                 return false;
             }
-            free((char*)causality);
```

Comment:
We have this same potential problem on two other places in this file, that we will need to fix.

The problem is that `const char* causality = "local";` on line 400 shall not be freed, which this PR fixes.
But in the case that `parseStringAttributeEzXml` actually parses a causality, then it should be deleted since it internally uses strdup. Otherwise there will be a minor memory leak.

We could check the return `bool` from `parseStringAttributeEzXml` to know it it should be freed or not.
Or maybe init the causality pointer to NULL, and treat NULL as "local" on line 414.

This could also be problem in other similar cases, not sure.

This is a comment meant for @robbr48, no need to fix in this PR